libs/core/langchain_core/messages/utils.py

```diff
@@ -373,6 +392,71 @@ def filter_messages(
         else:
             pass
 
+        if include_tool_names or exclude_tool_names:
+            if (
+                isinstance(msg, AIMessage)
+                and msg.tool_calls
+                or (
+                    isinstance(msg.content, list)
+                    and any(
+                        block.get("type") == "tool_use"
+                        for block in msg.content
+                        if isinstance(block, dict)
+                    )
+                )
+            ):
+                msg = msg.copy(deep=True)
+                filtered_tool_calls = []
+                for tc in msg.tool_calls:
+                    if exclude_tool_name_condition(tc["name"]):
+                        continue
+                    else:
+                        filtered_tool_calls.append(tc)
+                msg.tool_calls = filtered_tool_calls
+                if isinstance(msg.content, list):
+                    filtered_content = []
+                    for block in msg.content:
+                        if (
+                            isinstance(block, dict)
+                            and block.get("type") == "tool_use"
+                            and exclude_tool_name_condition(block.get("name"))
+                        ):
+                            continue
```

Comment:
is this for practical reasons (e.g., having inconsistent tool_use blocks and .tool_calls is problematic) or for aesthetics?